src/Modal/index.js

```diff
@@ -0,0 +1,266 @@
+      modalPortalElement.id = 'oio-modal-container'
+
+      // Append Modal Portal to OIO Container
+      // If no OIO Container exists, we append to body
+      const oioContainer = document.getElementById('oio-container') || document.body
+      oioContainer.appendChild(modalPortalElement)
+   }
+
+   const portal = document.getElementById('oio-modal-container')
+
+   // =======================================================
+   // Styles for Modal Window
+   // =======================================================
+
+   const windowProps = {
+      width,
+      height,
+      backgroundColor,
+      borderRadius,
+      boxShadow,
+      margin,
+      maxHeight,
+      maxWidth,
+      minHeight,
+      minWidth
+   }
+
+   // =======================================================
+   // Open and Close Events
+   // =======================================================
+
+   // When open animation has completed
+   const handleOpenComplete = () => {
+      if (onOpenComplete) {
+         onOpenComplete()
+      }
+   }
+
+   // When Close Animation has completed
+   const handleCloseComplete = () => {
+      modalIsActive.current = false
+      setIsClosing(false)
+      portal.removeAttribute('style')
+
+      if (onCloseComplete) {
+         onCloseComplete()
+      }
+   }
+
+   // Currently this is only triggered by clicking on overlay
+   // TODO: Handle if user presses escape key
+   const handleCloseTrigger = () => {
+      if (onCloseTrigger) {
+         onCloseTrigger()
+      }
+   }
+
+   // =======================================================
+   // Handle changes to open prop
+   // =======================================================
+
+   useEffect(() => {
+      if (open) {
+         // We need to ensure the portal container has the following style
+         // to ensure that the modal displays correctly and fills the entire screen
+         // Note: we ONLY want it to have this style when modal is open,
+         // otherwise it may cover content  while modal is closed
+         portal.setAttribute(
+            'style',
+            'position: fixed; top: 0; left: 0; right: 0; bottom: 0;'
+         )
+
+         if (onOpen) {
+            onOpen()
+         }
+
+      // If Modal open changes from true to false,
+      // begin close animation only if Modal was previously open
+      } else if (modalIsActive.current) {
+         setIsClosing(true)
+
+         if (onClose) {
+            onClose()
+         }
+      }
+
+      return () => {
+         portal.removeAttribute('style')
+      }
+   }, [open])
+
+   // Cleanup on unmount
+   useEffect(() => () => portal.remove(), [])
+
+   // =======================================================
+   // Render
+   // =======================================================
+
+   return ReactDOM.createPortal(
+      <View
+         display={modalIsActive.current ? 'flex' : 'none'}
+         css={{ pointerEvents: shouldPropagatePointerEvents ? 'none' : 'auto' }}
```

Comment:
Does this still apply to the Modal after we kinda discussed Notification being a separate component?